system-test/datastore.ts

```diff
@@ -811,6 +812,83 @@ describe('Datastore', () => {
       const [entities] = await datastore.runQuery(q);
       assert.strictEqual(entities!.length, 6);
     });
+    describe('with the filter function using the Filter class', () => {
+      it('should run a query with one property filter', async () => {
+        const filter = new PropertyFilter('family', '=', 'Stark');
+        const q = datastore
+          .createQuery('Character')
+          .filter(filter)
+          .hasAncestor(ancestor);
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 8);
+      });
+      it('should run a query with two property filters', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter(new PropertyFilter('family', '=', 'Stark'))
+          .filter(new PropertyFilter('appearances', '>=', 20));
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
+      });
+      it('should run a query using new Filter class with filter', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter('family', 'Stark')
+          .filter(new PropertyFilter('appearances', '>=', 20));
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
+      });
+      it('should run a query using an AND composite filter', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter(
+            AND([
+              new PropertyFilter('family', '=', 'Stark'),
+              new PropertyFilter('appearances', '>=', 20),
+            ])
+          );
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
```

Comment:
Is it possible to also assert that the resulting entities indeed have the values `family = Stark` and `apprearances >= 20`?

Reply:
I looked at the data and there is actually one entity that has family `['Stark', 'Tully']`. I believe this is intended behaviour for the filter because this is also the result when we test a case where legacy `.filter` is used.

Reply:
I just checked `should filter queries with defined indexes` on main. It seems this filter checks to see if `Stark` is present. I guess this matches what you said. We can check for its pressence.

Reply:
Done in last commit